### DISCUSSION TITLE
Point coordinates in angular position

### DISCUSSION THREAD

**Discussion by julien8745** [cncjs/cncjs#855] [Q&A] [answered]
Good morning,
I want to control an arm with two Z links via an arduino and GLBR. But GLBR is only programmed for linear axes and not angular connections.
Is it possible to configure CNCJS so that the coordinates to be reached are calculated via the inverse geometric model of my arm then the angles corresponding to the position of the point to be reached converted into "G01 Xangle Yangle" in the G-code?

**Comment by MitchBradley** [comment, marked as answer]
It is very unlikely that kinematics will be added to cncjs.  Kinematics transforms are not usually considered to be the responsibility of "sender" programs like cncjs.  You could do the kinematics transform before cncjs, preprocessing the gcode file so the coordinates that cncjs passes through to the sender are in cartesian space, or you could use a different controller firmware that supports kinematics modules, such as FluidNC or GrblHal.

**Comment by julien8745** [comment]
Okay thank you for your answer, I was starting to create a program allowing me to modify my gcode. I'm going to find out about GrblHal because I've never heard of it. My goal is to make a plasma cutting machine similar to an ArcDroid
